docs/docs/guides/advanced/support_additional_rpc_methods/index.md

```diff
@@ -0,0 +1,114 @@
+:::
+
+Following tutorial will guide you through the process of creating a custom plugin to extend the functionality of web3.js 4.x and add support for new RPC methods.
+
+#### Creating new RPC methods Plugin in 4 Steps
+
+1. First add web3.js as peer dependency in project´s `package.json` and create a TypeScript class for your plugin. This class should extend the `Web3Plugin` class provided by web3.js.
+
+:::info
+This will give your plugin access to [requestManager](/api/web3-core/class/Web3Context#requestManager) and [accountProvider](/api/web3-core/class/Web3Context#accountProvider).
+:::caution
+
+```ts
+import { Web3PluginBase } from 'web3';
+
+export default class CustomRpcMethodsPlugin extends Web3PluginBase {
+	// step 2
+	// ...
+}
+```
+
+2. After that add public `pluginNamespace` property. This will be used to access your plugin, as mentioned in step number 5 code example.
+
+```ts
+import { Web3PluginBase } from 'web3';
+
+export default class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods'; // step 3
+}
+```
+
+3. Once plugin class is created using above mentioned steps, its very easy to add new RPC methods like:
+
+```ts
+import { Web3PluginBase } from 'web3';
+
+export default class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods';
+
+	public async customRpcMethod() {
+		// step 4
+		return this.requestManager.send({
+			// plugin has access to web3.js internal features like request manager
+			method: 'custom_rpc_method',
+			params: [],
+		});
+	}
+}
+```
+
+4. Final step is setting up module [augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation), this will allow you to access plugin on web3 object.
+
+```ts
+import { Web3PluginBase } from 'web3';
+
+export default class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods';
+
+	public async customRpcMethod() {
+		// step 4
+		return this.requestManager.send({
+			// plugin has access to web3.js internal features like request manager
+			method: 'custom_rpc_method',
+			params: [],
+		});
+	}
+}
+
+// Module Augmentation
+declare module 'web3' {
+	// step 4
+
+	interface Web3Context {
+		customRpcMethods: CustomRpcMethodsPlugin;
+	}
+}
+```
+
+#### Using Web3 Custom PRC Plugin (with web3 instance)
+
+5. First add plugin in your plugin consumer project's `package.json`, create web3 and plugin instances, and after that use `.registerPlugin` method with some web3.js module (in following example its registered with main web3).
+
+Once plugin is registered its custom methods will be available to use.
+
+```ts
+import { Web3 } from 'web3';
+import CustomRpcMethodsPlugin from 'web3-plugin-example';
```

Comment:
I suggest importing plugin from a local folder like this
`import CustomRpcMethodsPlugin from './path/to/plugin';`, because importing from `'web3-plugin-example'` looks like a separate npm module and can be confusing for users.

Reply:
but normally plugins are supposed to be in independent repo and npm package.

Reply:
yes. if use some other plugin. but if the user just creates it inside some private project he needs to user it locally.
in this guide, we do not have information on how to deploy the plugin to the npm and make it public. That's why I think it will be better to use local import